uint8 type
string details
